tests/integration/testsuites/gateway/scenario.go

```diff
@@ -0,0 +1,279 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+	"github.com/cucumber/godog"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/helpers"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/manifestprocessor"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/resource"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/testcontext"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"log"
+	"path"
+	"time"
+)
+
+const manifestsPath = "testsuites/gateway/manifests/"
+
+type scenario struct {
+	testID          string
+	namespace       string
+	k8sClient       dynamic.Interface
+	resourceManager *resource.Manager
+	config          testcontext.Config
+}
+
+func initScenario(ctx *godog.ScenarioContext, ts *testsuite) {
+	scenario, err := createScenario(ts)
+
+	if err != nil {
+		log.Fatalf("could not initialize custom domain endpoint err=%s", err)
+	}
+
+	ctx.Step(`^there is an APIGateway operator in "([^"]*)" state$`, scenario.thereIsAnAPIGatewayOperator)
+	ctx.Step(`^there is a "([^"]*)" gateway in "([^"]*)" namespace$`, scenario.thereIsAGateway)
+	ctx.Step(`^there is a "([^"]*)" secret in "([^"]*)" namespace$`, scenario.thereIsACertificate)
+	ctx.Step(`^there is an "([^"]*)" APIRule$`, scenario.thereIsAnAPIRule)
+	ctx.Step(`^disabling kyma gateway will result in "([^"]*)" due to existing APIRule$`, scenario.gatewayErrorWhenKymaGatewayDisabled)
+	ctx.Step(`^enabling kyma gateway back will result in "([^"]*)" state$`, scenario.gatewayReadyWhenEnabled)
+	ctx.Step(`^removing an APIRule and APIGateway operator will also remove "([^"]*)" in "([^"]*)" namespace$`, scenario.removingAPIRuleUnblocksGatewayDeletion)
```

Comment:
It's the CR, not the operator.
```suggestion
	ctx.Step(`^removing an APIRule and APIGateway CR will also remove "([^"]*)" in "([^"]*)" namespace$`, scenario.removingAPIRuleUnblocksGatewayDeletion)
```